### PR TITLE
Add support for 3.8

### PR DESCRIPTION
**Pull request by naphta**
Just a nice quick change hopefully. I've ran tox and the tests passed but really I'm just making sure when used with 3.8 it no longer pulls in the version of mypy which depends on typed_ast as that's been removed in 3.8.

I'm not very familiar with tox so correct me if I've made an erroneous change to the tox.ini (I ran `tox -e py38` inside a 3.8-dev interpreter)

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -35,6 +35,7 @@ def read(fname):
         'pytest>=2.8; python_version>="3.5"',
         'mypy>=0.570,<0.700; python_version<"3.5"',
         'mypy>=0.570; python_version>="3.5"',
+        'mypy>=0.701; python_version>="3.8"',
```
[the remark this answers is not in the document]
Comment (naphta) [reply setup.py]: On a bit more testing (not sure how I didn't run into this yesterday) we need to support no less than 0.710 for it to sucessfully install mypy.

I'm having a play with the `python_version` and the only way I can get it to run on the pre-release builds is just by specifying the requires as starting at the first `b` release.

**Comment by naphta** [comment]
I've put a few of the suggested changes in (and looking at the ongoing travis build) I can't see any reason why it wouldn't pass.

File: changelog.md [hunk 2]
```diff
@@ -1,5 +1,8 @@
 # Changelog
 
+## 0.4.1
```
[the remark this answers is not in the document]
Comment (naphta) [reply changelog.md]: Dropped the changes back down to 0.4.0

**Comment by neozenith** [comment]
@naphta For that failing job in Travis:

`The job exceeded the maximum time limit for jobs, and has been terminated.`

It think the test matrix for py36 is too big now. May have to partition the matrix by python and pytest version or some other means.

**Comment by naphta** [comment]
> @naphta For that failing job in Travis:
> 
> `The job exceeded the maximum time limit for jobs, and has been terminated.`
> 
> It think the test matrix for py36 is too big now. May have to partition the matrix by python and pytest version or some other means.

I'm not hugely familiar with Travis so I'm just wondering if you can give me any guidance for doing so?

**Comment by dmtucker** [comment]
@naphta I suspect it's because of the 0.720 addition. Maybe we should add that in a separate PR.

File: setup.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (dmtucker) [reply setup.py]: 🤔 Hmmm... I was able to run 3.8 against 0.670 (but not 0.660). What error are you seeing?

[the remark this answers is not in the document]
Comment (naphta) [reply setup.py]: It's a C error with older versions of mypy which directly depend on typed_ast which was removed in 3.8. From what I can tell that change got merged into MyPy somewhere after 0.701, but I've had mixed results every time I've ran the tox tests now. 

I'll find the pull request where this is all discussed.

[the remark this answers is not in the document]
Comment (naphta) [reply setup.py]: https://github.com/python/mypy/pull/6539

[the remark this answers is not in the document]
Comment (dmtucker) [reply setup.py]: Note that the commit that PR merged (https://github.com/python/mypy/commit/69a5a0286e0129a71624001c38c2d4897b594574) is included in 0.700:
Screen Shot 2019-08-01 at 9 58 56 AM

[the remark this answers is not in the document]
Comment (naphta) [reply setup.py]: Oh! I never noticed that you can see the tags a commit / PR was included at. I was eye balling from the release dates.

I'll make that change next time I'm at a computer, and also remove the 720 release. I can open up the pull request for that separately too, I'll just need a bit of help possibly making the Travis changes.

**Comment by neozenith** [comment]
Happy to chase up adding 0.720 support and addressing the travis yml in issue #46 if you want to remove it from this PR.

**Comment by naphta** [comment]
> Happy to chase up adding 0.720 support and addressing the travis yml in issue #46 if you want to remove it from this PR.

All done.

**Comment by naphta** [comment]
Righteo, looking at the build now the errors you can see against 3.8 are what I needed to bump the version past 0.710 to solve. I'm not sure if there were other changes made to support 3.8 which fell outside of that pull request I mentioned earlier. I'm happy to bump the version up again but I'll have a look for any changes made between those releases which might be causing it.

I know there's an open issue to put in a version matrix in the documentation to make this a little bit clearer so speaking to whomever opened that issue might give me the answer.

**Comment by naphta** [comment]
> Righteo, looking at the build now the errors you can see against 3.8 are what I needed to bump the version past 0.710 to solve. I'm not sure if there were other changes made to support 3.8 which fell outside of that pull request I mentioned earlier. I'm happy to bump the version up again but I'll have a look for any changes made between those releases which might be causing it.
> 
> I know there's an open issue to put in a version matrix in the documentation to make this a little bit clearer so speaking to whomever opened that issue might give me the answer.

I'm still struggling to find wind of what makes the versions different but I did track down https://github.com/python/mypy/issues/7001 which suggests a build after the 16th of June would work, which sure enough v0.710 is just that. I did do a comparison of the two changesets and nothing specifically jumped out at me but I can't say I'm very familiar with their source code.

I think for now since 3.8 is a prerelease we can revisit the proper supported version once it drops for real if there's a problem.

**Comment by naphta** [comment]
Any reason not to merge now?

**Comment by dmtucker** [comment]
I think this is why 0.710 was needed: https://github.com/python/mypy/pull/6937
In other words, `mypy` 0.70x itself works with Python 3.8, but a bug in its packaging [requires a version of `typed-ast` that does not](https://github.com/python/mypy/blob/d40990f5b91b1f06918222d1041441b7ff63f79a/setup.py#L177)  (< `1.4.0`).

**Pull request merged**